packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -232,9 +235,11 @@ export class WorkspaceCommandContribution implements CommandContribution {
                         initialValue: vacantChildUri.path.base,
                         validate: name => this.validateFileName(name, parent, true)
                     }, this.labelProvider);
-                    dialog.open().then(name => {
+                    await dialog.open().then(async name => {
                         if (name) {
-                            this.fileSystem.createFolder(parentUri.resolve(name).toString());
+                            const folderUri = parentUri.resolve(name);
+                            this.commandEvents.fireNewFileCommand(folderUri);
+                            await this.fileSystem.createFolder(parentUri.resolve(name).toString());
```

Comment:
I think such event just should belong to `WorkspaceCommandContribution`, so here:
```ts
const folderUri = parentUri.resolve(name);
await this.fileSystem.createFolder(folderUri.toString());
this.onDidCreateNewFolderEmitter.fire(folderUri);
```

I don't think we need mention command and execute in the event name. I see such API:
```ts
export class WorkspaceCommandContribution {
   readonly onDidCreateNewFolder: Event<URI>; 
   createNewFolder(folderUri: URI): Promise<void>;
}
```

Clients should inject `WorkspaceCommandContribution` and then can call `createNewFolder` directly or listen when it happens.

Reply:
done